src/chart/speclayout/tablechart.h

```diff
@@ -18,14 +18,25 @@ template <typename Item>
 void TableChart::setupVector(std::vector<Item> &items,
     bool singleColumn)
 {
-	if (items.empty()) return;
-
 	auto size = 0;
 	for (auto &item : items)
 		if (item.enabled) ++size;
 
-	auto rowsize =
-	    singleColumn ? 1 : static_cast<ssize_t>(ceil(sqrt(size)));
+	if (size == 0) return;
+
+	ssize_t rowsize{1};
+	if (!singleColumn) {
+		rowsize = static_cast<ssize_t>(ceil(sqrt(size)));
+		auto sq = static_cast<ssize_t>(ceil(sqrt(
+		    16.0 * static_cast<double>(rowsize * rowsize) / 9.0)));
+
+		auto rem = rowsize - 1 - (size - 1) % rowsize;
```

Comment:
it would be more readable if this would be moved to a separate function with a descriptive name.